### ISSUE TITLE
`astro dev run dags list` results in a runtime error

### ISSUE THREAD

**Issue by RNHTTR** [astronomer/astro-cli#1358]
**Describe the bug**

Running `astro dev run dags list` results in a runtime error

```
astro dev run dags list
Running: airflow dags list
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x1046752ec]

goroutine 45 [running]:
bufio.(*Reader).Read(0x0, {0x14000662000?, 0x14000186200?, 0x1400006e640?})
	bufio/bufio.go:216 +0x9c
github.com/docker/docker/pkg/stdcopy.StdCopy({0x105746380, 0x14000124008}, {0x105746380, 0x14000124010}, {0x105743660, 0x0})
	github.com/docker/docker@v20.10.7+incompatible/pkg/stdcopy/stdcopy.go:108 +0x350
github.com/astronomer/astro-cli/docker.ExecPipe.func1()
	github.com/astronomer/astro-cli/docker/docker.go:42 +0x4c
created by github.com/astronomer/astro-cli/docker.ExecPipe
	github.com/astronomer/astro-cli/docker/docker.go:40 +0x180
```

**What CLI Version did you experience this bug?**

Astro CLI Version: 1.18.2

**This CLI bug is related to which Astronomer Platform?**

- [ ] Astro
- [ ] Software
- [x] Local development

**What Operating System is the above CLI installed on?**

**🪜 Steps To Reproduce**

Run `astro dev start` then `astro dev run dags list`

**Comment by afastronomer** [comment]
Other customer faced same issue with `astro dev run providers list`.

**Comment by nathadfield** [comment]
I think it is raised by any `astro dev run` command because even trying to do this returns the error.

```
astro dev run --help
```

Not sure what of automated tests you run, but I might suggest that they are not sufficient if they missed something like this.

**Comment by sunkickr** [comment]
on version 1.19.1 I am not able to reproduce:

<img width="495" alt="Screenshot 2023-09-01 at 3 13 50 PM" src="https://github.com/astronomer/astro-cli/assets/63181127/b1aafc39-0896-4b9a-a999-1e4690479f24">

**Comment by RNHTTR** [comment]
I'm able to reproduce with 1.19.1:

```
 % astro version
Astro CLI Version: 1.19.1
 % astro dev run --help
Running: airflow --help
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x104f35e18]

goroutine 69 [running]:
bufio.(*Reader).Read(0x0, {0x140003ca000?, 0x14000332c80?, 0x105fcc5c0?})
	bufio/bufio.go:223 +0x88
github.com/docker/docker/pkg/stdcopy.StdCopy({0x105fbc768, 0x14000112028}, {0x105fbc768, 0x14000112030}, {0x105fbcc28, 0x0})
	github.com/docker/docker@v20.10.7+incompatible/pkg/stdcopy/stdcopy.go:108 +0x34c
github.com/astronomer/astro-cli/docker.ExecPipe.func1()
	github.com/astronomer/astro-cli/docker/docker.go:42 +0x4c
created by github.com/astronomer/astro-cli/docker.ExecPipe in goroutine 1
	github.com/astronomer/astro-cli/docker/docker.go:40 +0x178
```

**Comment by sunkickr** [comment]
Temporary work around: install the latest version the CLI with curl instead of brew: `sudo bash < <(curl -sSL https://install.astronomer.io/)`

**Comment by sunkickr** [comment]
Another work around is to install with `brew install astronomer/tap/astro` instead of `brew install astro`. Something is wrong with the binary hosted by the brew tap.

**Comment by sunkickr** [comment]
This is fixed in v1.19.2
